dashboard/src/modules/components/ProfileComponent/index.jsx

```diff
@@ -0,0 +1,243 @@
+import React, { useEffect, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import {
+  TextContent,
+  Text,
+  TextVariants,
+  Grid,
+  GridItem,
+  Card,
+  Level,
+  LevelItem,
+  Button,
+  TextInput,
+  CardBody,
+} from "@patternfly/react-core";
+import {
+  UserAltIcon,
+  PencilAltIcon,
+  KeyIcon,
+  RedoIcon,
+} from "@patternfly/react-icons";
+import "./index.less";
+import avatar from "assets/images/avatar.jpg";
+import {
+  getProfileDetails,
+  updateUserDetails,
+  sendForUpdation,
+  resetUserDetails,
+} from "actions/profileActions";
+
+const ProfileComponent = () => {
+  const [editView, setEditView] = useState(false);
+  const dispatch = useDispatch();
+
+  const user = useSelector((state) => state.userProfile.userDetails);
+  const loginDetails = useSelector((state) => state.userAuth.loginDetails);
+
+  const edit = () => {
+    setEditView(!editView);
+  };
+  const saveEdit = () => {
+    const isDone = dispatch(sendForUpdation());
+    if(isDone) setEditView(false);
+  };
+  useEffect(() => {
+    if (loginDetails?.username) {
+      dispatch(getProfileDetails());
+    }
+  }, [dispatch, loginDetails?.username]);
+
+  const formatData = (registered) => {
+    const registeredDate = new Date(registered);
+    let mon = registeredDate?.getMonth() + 1;
+    mon = mon < 10 ? "0" + mon : mon;
+    return (
+      mon +
+      "/" +
+      registeredDate?.getDate() +
+      "/" +
+      registeredDate?.getFullYear()
+    );
```

Comment:
Would something like `Date.prototype.toLocaleTimeString()` be better here? I don't really like the idea of a fixed hardcoded date format. Even though this is "US standard" it's going to look odd to a lot of people... and I have always preferred the YYYY-MM-DD format myself because it sorts naturally. If we use the built-in localization mechanisms, that's one less detail to worry about.